ts/third_party/sourcemapper/sourcemapper.ts

```diff
@@ -0,0 +1,265 @@
+        sourceMap.RawSourceMap;
+  } catch (e) {
+    throw new Error(
+        'An error occurred while reading the ' +
+        'sourcemap file ' + mapPath + ': ' + e);
+  }
+
+  /*
+   * If the sourcemap file defines a "file" attribute, use it as
+   * the output file where the path is relative to the directory
+   * containing the map file.  Otherwise, use the name of the output
+   * file (with the .map extension removed) as the output file.
+   */
+  const outputBase =
+      consumer.file ? consumer.file : path.basename(mapPath, '.map');
+  const parentDir = path.dirname(mapPath);
+  const outputPath = path.normalize(path.join(parentDir, outputBase));
+
+  const sources = Array.prototype.slice.call(consumer.sources)
+                      .filter((value: string) => {
+                        // filter out any empty string, null, or undefined
+                        // sources
+                        return !!value;
+                      })
+                      .map((relPath: string) => {
+                        // resolve the paths relative to the map file so that
+                        // they are relative to the process's current working
+                        // directory
+                        return path.normalize(path.join(parentDir, relPath));
+                      });
+
+  if (sources.length === 0) {
+    throw new Error('No sources listed in the sourcemap file ' + mapPath);
+  }
+  sources.forEach((src: string) => {
+    infoMap.set(
+        path.normalize(src),
+        {outputFile: outputPath, mapFile: mapPath, mapConsumer: consumer});
+  });
+}
+
+export class SourceMapper {
+  infoMap: Map<string, MapInfoInput>;
+
+  /**
+   * @param {Array.<string>} sourcemapPaths An array of paths to .map sourcemap
+   *  files that should be processed.  The paths should be relative to the
+   *  current process's current working directory
+   * @param {Logger} logger A logger that reports errors that occurred while
+   *  processing the given sourcemap files
+   * @constructor
+   */
+  constructor() {
+    this.infoMap = new Map();
+  }
+
+  /**
+   * Used to get the information about the transpiled file from a given input
+   * source file provided there isn't any ambiguity with associating the input
+   * path to exactly one output transpiled file.
+   *
+   * @param inputPath The (possibly relative) path to the original source file.
+   * @return The `MapInfoInput` object that describes the transpiled file
+   *  associated with the specified input path.  `null` is returned if either
+   *  zero files are associated with the input path or if more than one file
+   *  could possibly be associated with the given input path.
+   */
+  private getMappingInfo(inputPath: string): MapInfoInput|null {
+    if (this.infoMap.has(path.normalize(inputPath))) {
+      return this.infoMap.get(inputPath) as MapInfoInput;
+    }
+
+    // TODO(#268): Remove commented section of code.
```

Comment:
Maybe remove this code now? Submitting commented code should generally be avoided.

Reply:
Removed the commented code.